src/layout/core/borders/container_draw.rs

```diff
@@ -33,21 +33,96 @@ impl ContainerDraw {
         x = gap / 2.0;
         w -= gap;
 
-        // Draw background of title bar
-        self.base.set_source_rgb(1.0, 0.0, 0.0);
-        self.base.set_color_source(title_color);
-        self.base.rectangle(x, 0.0, w, title_size);
-        self.base = try!(self.base.check_cairo());
-        self.base.fill();
-        self.base = try!(self.base.check_cairo());
-
-        // Draw title text
-        self.base.move_to(title_x, title_y);
-        self.base = try!(self.base.check_cairo());
-        self.base.set_color_source(title_font_color);
-        self.base = try!(self.base.check_cairo());
-        self.base.show_text(title.as_str());
-        self.base = try!(self.base.check_cairo());
+        // TODO: This shouldn't be clone for performance reasons, but self
+        // needs to not be borrowed later with all the paint operations, and
```

Comment:
Hmm, yes this is tricky, but I think can see a way around it:

All you care about is which index is the active one, so if `childrens` exist[1] you just loop through and get the index that is the active one (which will copy out that little `usize`) and use that in the match statements. No more needless copying and now your match statements have less duplicate code as a bonus.


[1] and it shouldn't unless there's a case where a container can be not tabbed/stacked and still have children, in which case this is unnecessary useless work. Though you can always hide it behind a lambda that caches, though that makes it so much more complicated it wouldn't be worth it and you should instead just do the needless calculation